modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1558,16 +1559,56 @@ private void _validateListTypeEntriesValues(
 		}
 	}
 
+	private void _validateObjectFieldIntegerTypeSize(
+			Map.Entry<String, Serializable> entry)
+		throws ObjectEntryValuesException {
+
+		Serializable entryValue = entry.getValue();
+
+		String entryValueString = entryValue.toString();
+
+		if (!entryValueString.isEmpty()) {
+			int value = GetterUtil.getInteger(entryValue);
+
+			if (!StringUtil.equals(String.valueOf(value), entryValueString)) {
+				throw new ObjectEntryValuesException.IntegerSizeExceeded();
+			}
+		}
+	}
+
+	private void _validateObjectFieldLongType(
+			Map.Entry<String, Serializable> entry)
+		throws ObjectEntryValuesException {
+
+		Serializable entryValue = entry.getValue();
+
+		String entryValueString = entryValue.toString();
+
+		if (!entryValueString.isEmpty()) {
+			long value = GetterUtil.getLong(entryValue);
+
+			if (!StringUtil.equals(
+					String.valueOf(value), entryValue.toString())) {
+
+				throw new ObjectEntryValuesException.LongMaxSizeExceeded();
```

Comment:
Could you create a generic message for this case, please?

Reply:
Sure!
Thanks for the tip!